### ISSUE TITLE
New component: Apache Flink Metric Receiver

### ISSUE THREAD

**Issue by JonathanWamsley** [open-telemetry/opentelemetry-collector-contrib#9715]
## The purpose and use-cases of the new component
This Apache Flink Metric Receiver will use the Flink [REST API](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/ops/metrics/#rest-api-integration) to collect Jobmanager, Taskmanager, Job, Task and Operator metrics.

Targeting versions 1.13.6, 1.14.4
The current latest stable release is 1.14.4 and Flink supports the current and previous minor release with bugfixes.

## Example configuration for the component

The following configuration settings are optional:
    - collection_interval: interval to collect metrics (default 10s)
    - endpoint: endpoint for the Flink REST API (default https://localhost:8081)
    - TLS settings based on OTEL HTTPClientSettings

```yaml
receivers:
  flinkmetrics:
    collection_interval: 10s
    endpoint: http://localhost:8081
    tls:
        insecure_skip_verify: true
```

## Telemetry data types supported

Metrics

**Comment by bogdandrutu** [comment]
I would suggest to name it "flinkmetrics" to be consistent with kafka, if there is any chance that we may have a receiver that receives telemetry via "flink" (not an expert, seems unlikely)

**Comment by djaglowski** [comment]
I'll sponsor this.

**Comment by djaglowski** [comment]
Closed by #10121
